<launch>
    <node pkg="gimbal_track" type="gimbal_track_state_publisher" name="gimbal_track_state_publisher" output="screen">
    </node>

      <node pkg="gimbal_track" type="gimbal_track_velocity_controller" name="gimbal_track_velocity_controller" output="screen">
    </node>

    <node ns="gimbal_track" name="gimbal_track_pid_pitch_controller" pkg="pid" type="controller" output="screen" >
      <param name="node_name" value="pitch_controller" />
      <param name="Kp" value="3.0" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="0.2" />
      <param name="upper_limit" value="180" />
      <param name="lower_limit" value="-180" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamhku/gimbal_track/setpoint_pitch" />
      <param name="topic_from_plant" value="/teamhku/gimbal_track/pitch_state" />
      <param name="topic_from_controller" value="/teamhku/gimbal_track/gimbal_control_effort_pitch" />
     </node>

    <node ns="gimbal_track" name="gimbal_track_yaw_controller" pkg="pid" type="controller" output="screen" >
      <param name="node_name" value="yaw_controller" />
      <param name="Kp" value="3" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="0.2" />
      <param name="upper_limit" value="180" />
      <param name="lower_limit" value="-180" />
      <param name="windup_limit" value="100" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamhku/gimbal_track/setpoint_yaw" />
      <param name="topic_from_plant" value="/teamhku/gimbal_track/yaw_state" />
      <param name="topic_from_controller" value="/teamhku/gimbal_track/gimbal_control_effort_yaw" />
    
     </node>

    <node name="gimbal_track_setpoint" pkg="gimbal_track" type="gimbal_track_setpoint" output="screen" />
<!-- 
   <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
      <rosparam command="delete" param="/diagnostic_aggregator" />
      <rosparam command="load" file="$(find pid)/config/pid_diags.yaml" />
   </node>

    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/teamhku/velocity_control_effort_x /teamhku/tag_x /teamhku/velocity_control_effort_y /teamhku/tag_y /setpoint/data" /> -->


</launch>
